### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by kozhukhovsky**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,3 @@
- .idea/*
+.idea/*
 *.iml
 .target/*
```
Comment (Sofasmile) [inline .gitignore]: add a blank line

File: pom.xml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+                    <configuration>
+                        <source>${jdk.version}</source>
+                        <target>${jdk.version}</target>
+                        <encoding>${project.build.sourceEncoding}</encoding>
+                    </configuration>
+                </plugin>
+            </plugins>
+        </pluginManagement>
+    </build>
+</project>
```
Comment (Sofasmile) [inline pom.xml]: add a blank line

File: src/main/java/mate/academy/internetshop/dao/impl/BucketDaoImpl.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    @Override
+    public Bucket deleteById(Long id) {
+        Bucket deletedBucket = get(id);
+        Storage.buckets.removeIf(bucket -> bucket.getId().equals(id));
+        return deletedBucket;
+    }
+
+    @Override
+    public Bucket deleteByBucket(Bucket bucket) {
+        get(bucket.getId());
```
Comment (Sofasmile) [inline src/main/java/mate/academy/internetshop/dao/impl/BucketDaoImpl.java]: do we need this line?

File: src/main/java/mate/academy/internetshop/dao/impl/ItemDaoImpl.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    @Override
+    public Item deleteById(Long id) {
+        Item deletedItem = get(id);
+        Storage.items.removeIf(item -> item.getId().equals(id));
+        return deletedItem;
+    }
+
+    @Override
+    public Item deleteByItem(Item item) {
+        get(item.getId());
```
Comment (Sofasmile) [inline src/main/java/mate/academy/internetshop/dao/impl/ItemDaoImpl.java]: the same

File: src/main/java/mate/academy/internetshop/dao/impl/OrderDaoImpl.java [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    @Override
+    public Order deleteById(Long id) {
+        Order deletedOrder = get(id);
+        Storage.orders.removeIf(order -> order.getId().equals(id));
+        return deletedOrder;
+    }
+
+    @Override
+    public Order deleteByOrder(Order order) {
+        get(order.getId());
```
Comment (Sofasmile) [inline src/main/java/mate/academy/internetshop/dao/impl/OrderDaoImpl.java]: :)

File: src/main/java/mate/academy/internetshop/dao/impl/UserDaoImpl.java [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+    @Override
+    public User deleteById(Long id) {
+        User deletedUser = get(id);
+        Storage.users.removeIf(user -> user.getId().equals(id));
+        return deletedUser;
+    }
+
+    @Override
+    public User deleteByUser(User user) {
+        get(user.getId());
```
Comment (Sofasmile) [inline src/main/java/mate/academy/internetshop/dao/impl/UserDaoImpl.java]: :)

File: src/main/java/mate/academy/internetshop/lib/Injector.java [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+
+public class Injector {
+    public static void injectDependency() throws IllegalAccessException {
+        Class[] classes = new Class[]{
+            BucketServiceImpl.class, ItemServiceImpl.class, OrderServiceImpl.class,
+            UserServiceImpl.class, Main.class
+        };
+
+        for (Class clazz : classes) {
+            for (Field field : clazz.getDeclaredFields()) {
```
Comment (Sofasmile) [inline src/main/java/mate/academy/internetshop/lib/Injector.java]: we can create two method:
first of them will get as parameter class or fields and make inject
second will call first method in the `for loop` for all classes or fields

File: src/main/java/mate/academy/internetshop/model/Item.java [hunk 8]
```diff
@@ -0,0 +1,28 @@
+package mate.academy.internetshop.model;
+
+import mate.academy.internetshop.lib.IdGenerator;
+
+public class Item {
+    private final Long id;
+    private final String name;
```
Comment (Sofasmile) [inline src/main/java/mate/academy/internetshop/model/Item.java]: better doesn't make this field as final because how we will update this object?

Comment (Sofasmile) [reply src/main/java/mate/academy/internetshop/model/Item.java]: the same with other fields

File: src/main/java/mate/academy/internetshop/service/BucketService.java [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+
+    Bucket deleteById(Long id);
+
+    Bucket deleteByBucket(Bucket bucket);
+
+    Bucket addItem(Long bucketId, Long itemId);
+
+    Bucket clear(Long bucketId);
+
+    List getAllItems(Long bucketId);
```
Comment (Sofasmile) [inline src/main/java/mate/academy/internetshop/service/BucketService.java]:
```suggestion
    List<Item> getAllItems(Long bucketId);
```

File: src/main/java/mate/academy/internetshop/service/UserService.java [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
+
+    User get(Long id);
+
+    User update(User user);
+
+    User deleteById(Long id);
+
+    User deleteByUser(User user);
+
+    List getOrders(Long userId);
```
Comment (Sofasmile) [inline src/main/java/mate/academy/internetshop/service/UserService.java]:
```suggestion
    List<Order> getOrders(Long userId);
```

File: src/main/java/mate/academy/internetshop/dao/impl/BucketDaoImpl.java (hunk 3 above)
Comment (kozhukhovsky) [reply src/main/java/mate/academy/internetshop/dao/impl/BucketDaoImpl.java]: Нет, но Богдан так в видео делал, чтобы проверить, что у нас существует Bucket. 
Я просто повторил :)

File: src/main/java/mate/academy/internetshop/lib/Injector.java [hunk 11]
```diff
@@ -25,34 +25,38 @@ public static void injectDependency() throws IllegalAccessException {
...
-                        field.set(null, Factory.getOrderService());
-                    }
-                    if (type == UserService.class) {
-                        field.set(null, Factory.getUserService());
-                    }
+            injectDependencyForClass(clazz);
+        }
+    }
+
+    private static void injectDependencyForClass(Class<?> clazz) throws IllegalAccessException {
```
Comment (Sofasmile) [inline src/main/java/mate/academy/internetshop/lib/Injector.java:32]: suggestion: think about optimization

File: src/main/java/mate/academy/internetshop/service/BucketService.java [hunk 12]
```diff
@@ -18,5 +19,5 @@
 
     Bucket clear(Long bucketId);
 
-    List getAllItems(Long bucketId);
+    List<Item> getAllItems(Long bucketId);
```
Comment (Sofasmile) [inline src/main/java/mate/academy/internetshop/service/BucketService.java:22]: the same in the realization

**Review by Sofasmile — APPROVED** [review]
make small fixes

**Pull request merged**